### PR TITLE
MINOR: Replace get_class($this) with static::class

### PR DESCRIPTION
**Pull request by sminnee**
Thanks @assertchris for the recommendation

### CODE REVIEW

File: code/Model/ErrorPage.php [hunk 1]
```diff
@@ -122,7 +122,7 @@ public function requireDefaultRecords()
 		parent::requireDefaultRecords();
 
 		// Only run on ErrorPage class directly, not subclasses
-		if (get_class($this) !== 'SilverStripe\\CMS\\Model\\ErrorPage' || !SiteTree::config()->create_default_pages) {
+		if (static::class !== 'SilverStripe\\CMS\\Model\\ErrorPage' || !SiteTree::config()->create_default_pages) {
```
Comment (assertchris) [inline code/Model/ErrorPage.php]:
```php
use SilverStripe\CMS\Model\ErrorPage;

// ...

if (static::class !== ErrorPage::class)  {
```

File: code/Model/SiteTree.php [hunk 2]
```diff
@@ -2231,7 +2231,7 @@ public function getCMSActions() {
 		$live = Versioned::get_one_by_stage('SilverStripe\\CMS\\Model\\SiteTree', Versioned::LIVE, array(
 			'"SiteTree"."ID"' => $this->ID
 		));
-		$infoTemplate = SSViewer::get_templates_by_class(get_class($this), '_Information', __CLASS__);
+		$infoTemplate = SSViewer::get_templates_by_class(static::class, '_Information', __CLASS__);
```
Comment (assertchris) [inline code/Model/SiteTree.php]: Or `SSViewer::get_templates_by_class(static::class, '_Information', self::class);`

File: code/Model/SiteTree.php [hunk 3]
```diff
@@ -1513,7 +1513,7 @@ public function requireDefaultRecords() {
 		parent::requireDefaultRecords();
 
 		// default pages
-		if(get_class($this) == __CLASS__ && $this->config()->create_default_pages) {
+		if(static::class == __CLASS__ && $this->config()->create_default_pages) {
```
Comment (assertchris) [inline code/Model/SiteTree.php]: `static::class == __CLASS__` or `static::class == self::class`

Comment (tractorcow) [inline code/Model/SiteTree.php]: 👍

**Comment by sminnee** [comment]
Can anyone think of a place where `__CLASS__` could not be replaced with `self::class`? I only code by global search & replace.

**Comment by assertchris** [comment]
According to docs, `::class` is FQCN. `__CLASS__` is FQCN where defined (so semantically the same as `self`, which is not changed by inheritance context. I can't think of a reason not to use `static::class` and `self::class` instead of `static::class` (to replace `get_class($this)`) and `__CLASS__`.

File: code/Model/ErrorPage.php (hunk 1 above)
Comment (sminnee) [inline code/Model/ErrorPage.php]: Actually `static::class !== self::class` can be used here

**Comment by sminnee** [comment]
Force pushed an amended commit.

**Comment by assertchris** [comment]
For reference:
screen shot 2016-09-28 at 11 56 39 am

**Comment by kinglozzer** [comment]
Good stuff 👍

**Pull request merged**